codegen/reconcile/main.go

```diff
@@ -295,11 +307,11 @@ func Reconcile{{ .APIVersionPrefix }}{{ .ResourceNamePlural }}(ctx context.Conte
 		}
 
 		if err := EnsureNamedObject(ctx, types.NamespacedName{Namespace: namespace, Name: name}, createObject, client, &{{ .ImportAlias }}.{{ .ResourceName }}{}, {{ .RequiresRecreate}}); err != nil {
-			return fmt.Errorf("failed to ensure {{ .ResourceName }} %s/%s: %v", namespace, name, err)
+			return fmt.Errorf("failed to ensure {{ .APIVersionPrefix }}{{ .ResourceName }} %s/%s: %v", namespace, name, err)
 		}
-	}
 
-	return nil
+		return nil
+	}
```

Comment:
Nothing is changed in this file, here except adding a closure and increasing the call-stack, so what's the point of this change?

My guess it's for using the `reconciling.TaskFn`, since generated funcs conform the signature, but they are never used in this case, so I propose to ether use them like they should be used as `reconciling.TaskFn`, or for now simply drop that codegen change.

Reply:
This change reduces the impedance mismatch between the functions we use to define the resources and this lib. We can definitely defer this to the PR where this is firstly used, but I thought that this gives a hint on how this will be used.

Note I could have implemented this without adding a function call but in our context the impact is negligible, and I preferred to avoid repetitions.